scripts/find_unique_isoform.py

```diff
@@ -128,64 +172,83 @@ def compareFunc(seqDict, outFile, minPercent):
                 # intersect to obtain unique isoforms in this sample
-                unique = set(this).difference(set(another))
+                unique = sorted(list( set(this).difference(set(another)) ))
 
                 # N-W alignment stats
                 for u in unique:
 
                     # all condidate sequences for N-W alignment
-                    allSeqs, allNames = [], []
-                    for s,d in allSamples.items():
-                        allSeqs += [ d[q]['seq'] for q in d.keys() if d[q]['seq'] != u]
-                        allNames += [ s+'_'+q for q in d.keys() if d[q]['seq'] != u ]
+                    condidateSeqs = [ s.seq for s in isoforms if s.seq != u ]
+                    condidateContigs = [ s.contig for s in isoforms if s.seq != u ]
+                    condidateStarts = [ s.start for s in isoforms if s.seq != u ]
+                    condidateNames = [ s.name for s in isoforms if s.seq != u ]
 
                     # N-W alignment and save for quick lookup
-                    for a in allSeqs:
+                    for a in condidateSeqs:
                         if (u,a) not in lookup: lookup[(u,a)] = edlib.align(u, a, mode = 'NW', task = 'path')
 
                     # alignment stats
-                    alns = [ lookup[(u,a)] for a in allSeqs ]
+                    alns = [ lookup[(u,a)] for a in condidateSeqs ]
                     eds = [ round(aln['editDistance']/len(u),2) for aln in alns ]
                     cigars = [ aln['cigar'] for aln in alns ]
 
                     # select alignments by percentile edit distance
                     if alns:
                         cut = numpy.percentile(eds, 100 - minPercent)
-                        picks = [ '__'.join([str(ed),allNames[i],allSeqs[i],cigars[i]]) for i,ed in enumerate(eds) if ed <= cut ]
+                        picks = [ '__'.join([str(ed),condidateNames[i],condidateContigs[i],str(condidateStarts[i]),condidateSeqs[i],cigars[i]]) for i,ed in enumerate(eds) if ed <= cut ]
                     else:
                         picks = []
 
-                    # output
-                    for qname,v in sampleDict.items():
-                        temp = [total, qname, sample, anotherSample, v['start'], u, ','.join(picks)]
-                        if v['seq'] == u: out.write('\t'.join( map(str, list(locus)+temp) ) +'\n')
+                    # output all isoforms in this sample that has sequence "u"
+                    for isoform in [ s for s in isoforms if s.sample == sample ]:
+                        temp = [total, isoform.name, sample, anotherSample, isoform.start, u, ','.join(picks)]
+                        if isoform.seq == u: outUni.write('\t'.join( map(str, list(locus)+temp) ) +'\n')
+
+
+        # all isoform identical/divergent record
+        for isoform in isoforms:
+
+            identical,divergent = [],[]
+
+            for s in isoforms:
+                if s.id == isoform.id: continue
```

Comment:
It took me some time to figure this out. I get it:

```
for s in isoforms:
    if s.id == isoform.id: 
        if s.seq == isoform.seq:
            identical.append(s.sample+'_'+s.name)
        else:
            divergent.append(s.sample+'_'+s.name)
````

Fair enough; I guess I try to avoid `continue` as it gets confusing

Reply:
I think here if `s.id == isoform.id`  the program skips the next if statement to the next `s in isoforms`. 
So it is 
```
for s in isoforms:
    if s.id != isoform.id: 
        if s.seq == isoform.seq:
            identical.append(s.sample+'_'+s.name)
        else:
            divergent.append(s.sample+'_'+s.name)
```

Reply:
Oh, right...

Uff, I hate `continue`. 

Well, if that's the logic, then we could rewrite the above for more clarify :) CC @BidaGu